projects/plugins/jetpack/class.jetpack-gutenberg.php

```diff
@@ -440,6 +440,34 @@ public static function is_gutenberg_available() {
 		return true;
 	}
 
+	/**
+	 * Check if the Newsletter feature should be enabled
+	 *
+	 * @since $$next_version$$
+	 *
+	 * @return bool
+	 */
+	public static function is_newsletter_enabled() {
```

Comment:
Great point, I agree that this could be an approach as well. 

Ultimately I went with this approach due to the task description in https://github.com/Automattic/jetpack/issues/29204

> We should not be showing the newsletter access level selector in posts if:
> 1. You don't have Stripe configured
> 2. You don't have any "Newsletter" plan configured 

We are currently overriding this filter on WPCOM. If we were to introduce this logic, the UI and backend code would be loaded, but the user wouldn't be able to use the functionality due to Stripe not being connected and newsletter plans not being configured.

I definitely see room for improvement in the future. Ultimately, in the future I see a UX where we prompt the user to complete Stripe KYC and configure newsletter plans

Reply:
> We are currently overriding this filter on WPCOM. If we were to introduce this logic, the UI and backend code would be loaded, but the user wouldn't be able to use the functionality due to Stripe not being connected and newsletter plans not being configured.

I disagree about that... Let me know if I am wrong, but this is the WPCOM override:
fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Szrzorefuvcf%2Qybnqre.cuc%236-og

And it would not change anythign to use my version, as it is basically the exact same logic as yours...

Reply:
Yup,  that's the override. In this case, all proxied users would have this feature enabled. I think this is not the behavior we want. 

cc: @artpi  for clarification.

Reply:
Yes, now I am sure you have misunderstood my approach.